src/game_player.cpp

```diff
@@ -712,17 +712,65 @@ bool Game_Player::CanWalk(int x, int y) {
 void Game_Player::BeginMove() {
 	int terrain_id = Game_Map::GetTerrainTag(GetX(), GetY());
 	const RPG::Terrain* terrain = ReaderUtil::GetElement(Data::terrains, terrain_id);
+	bool red_flash = false;
+
+	// Step Damage for States:
+	for (auto hero : Main_Data::game_party->GetActors()) {
+		for (auto state_id : hero->GetInflictedStates()) {
+			RPG::State *state = ReaderUtil::GetElement(Data::states, state_id);
```

Comment:
This whole loop logic could be in a Game_Actor::ApplyStateDamage function.
About the state steps counter: Is this really global for all actors? I mean when one hero is poisoned and another is poisoned one step later, do they receive damage at the same time?